docling_core/types/doc/document.py

```diff
@@ -1321,6 +1480,7 @@ def export_to_markdown(  # noqa: C901
         indent: int = 4,
         text_width: int = -1,
         page_no: Optional[int] = None,
+        image_dir: Optional[Path] = None,
```

Comment:
It is not a proper solution to provide a path argument for the `image_dir`. This would mean:
- The `image_dir` is being written to by the export method straight
- The exported document representation (i.e. Dict, MD string, HTML string) is returned by the export method but not written to the same place, and you make it the users responsibility to then save the exported object where the images are.